packages/langium/test/parser/langium-parser-builder.test.ts

```diff
@@ -259,6 +259,42 @@ describe('Parser calls value converter', () => {
     });
 });
 
+describe('Parsing with Inferred Types', () => {
+
+    const services = createLangiumGrammarServices();
+    const parser = parseHelper<Grammar>(services.grammar);
+
+    test('verify no TypeError for InferredType in scope', async () => {
+        const c = `
+        type T = A;
```

Comment:
Where is A defined?

Reply:
That's a good point! It's not defined in this case. It can be added, however this alone was enough to throw a type error. It could be defined for this case, but the result would be the same.

Reply:
Ah! Semantic checks come later, I guess ^^*. It is just a little bit confusing to see a broken example passing the test

Reply:
Yeah it is a little strange still 😅 . I've seen some other tests that do something similar, dropping the `ID` terminal or something like that.